### PR TITLE
refactor: Share some models with the web project

### PR DESCRIPTION
**Pull request by BenjaminMichaelis**

### CODE REVIEW

File: Directory.Packages.props [hunk 1]
```diff
@@ -3,11 +3,13 @@
     <ManagePackageVersionsCentrally>true</ManagePackageVersionsCentrally>
     <CentralPackageTransitivePinningEnabled>false</CentralPackageTransitivePinningEnabled>
     <AccessToNugetFeed>true</AccessToNugetFeed>
+    <ToolingPackagesVersion>1.1.1.3883</ToolingPackagesVersion>
   </PropertyGroup>
   <ItemGroup Condition="$(AccessToNugetFeed)">
-    <PackageVersion Include="ContentFeedNuget" Version="1.1.0-3867" />
+    <PackageVersion Include="ContentFeedNuget" Version="$(ToolingPackagesVersion)" />
```
Comment (Keboo) [inline Directory.Packages.props:9]: Doesn't this break dependabot updates?

File: EssentialCSharp.Web/Extensions/FileInfoExtensions.cs [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+            return null;
+        }
+
+        // Read the JSON file
+        string jsonString = File.ReadAllText(guidelinesJsonFile.FullName);
+
+        // Deserialize the JSON string into a List<GuidelineListing>
+        List<GuidelineListing>? guidelines = JsonSerializer.Deserialize<List<GuidelineListing>>(jsonString, GuidelineListing.Options);
+
+        if (guidelines != null && guidelines.Count > 0)
```
Comment (Keboo) [inline EssentialCSharp.Web/Extensions/FileInfoExtensions.cs:22]:
```suggestion
        if (guidelines?.Count > 0)
```

File: EssentialCSharp.Web/Extensions/StringExtensions.cs [hunk 3]
```diff
@@ -65,4 +65,17 @@ public static string KeyToHeading(this string str)
     {
         return System.Globalization.CultureInfo.CurrentCulture.TextInfo.ToTitleCase(str.Trim().ToLowerInvariant().Replace('-', ' '));
     }
+
+    public static SiteMapping? Find(this string key, IList<SiteMapping> siteMappings)
+    {
+        key ??= siteMappings[0].Key;
```
Comment (Keboo) [inline EssentialCSharp.Web/Extensions/StringExtensions.cs:71]: Is there a case where we are concerned that `key` comes in `null`?

File: EssentialCSharp.Web/Extensions/StringExtensions.cs [hunk 4]
```diff
@@ -65,4 +65,17 @@ public static string KeyToHeading(this string str)
     {
         return System.Globalization.CultureInfo.CurrentCulture.TextInfo.ToTitleCase(str.Trim().ToLowerInvariant().Replace('-', ' '));
     }
+
+    public static SiteMapping? Find(this string key, IList<SiteMapping> siteMappings)
```
Comment (Keboo) [inline EssentialCSharp.Web/Extensions/StringExtensions.cs:69]: Having this method on `string` feels a little off. What about inverting the order of the parameters and making the `this` parameter `IList<SiteMapping> siteMappings`? Perhaps renaming it to "FindByKey" or similar?

File: Directory.Packages.props (hunk 1 above)
Comment (BenjaminMichaelis) [reply Directory.Packages.props:9]: I don't actually think so, I had something in another project (can't remember where) where this seemed to work. But I guess we will find out? 

If it doesn't I'll do dependabot grouped updates, and add a msbuild flag to fail the build if these get out of sync

File: EssentialCSharp.Web/Extensions/StringExtensions.cs (hunk 3 above)
Comment (BenjaminMichaelis) [reply EssentialCSharp.Web/Extensions/StringExtensions.cs]: Good catch

**Review by Keboo — APPROVED** [review]

**Pull request merged**